cirq/study/sweepable.py

```diff
@@ -16,35 +16,44 @@
 
 from typing import Dict, Iterable, Iterator, List, Union, cast
 import itertools
+import sympy
 
 from cirq._doc import document
 from cirq.study.resolver import ParamResolver, ParamResolverOrSimilarType
 from cirq.study.sweeps import ListSweep, Points, Sweep, UnitSweep, Zip
 
 
-Sweepable = Union[Dict[str, float], ParamResolver, Sweep, Iterable[
-    Union[Dict[str, float], ParamResolver, Sweep]], None]
+SweepableDictType = Dict[Union[str, sympy.Symbol], Iterable[
+    Union[float, str, sympy.Basic]]]
+document(
+    SweepableDictType,  # type: ignore
+    """Dictionary from symbols to sequence of values taken.""")
+
+Sweepable = Union[
+    ParamResolverOrSimilarType, Sweep, SweepableDictType, Iterable[
+        Union[ParamResolverOrSimilarType, Sweep, SweepableDictType]], None]
 document(
     Sweepable,  # type: ignore
     """An object or collection of objects representing a parameter sweep.""")
 
 
 def to_resolvers(sweepable: Sweepable) -> Iterator[ParamResolver]:
     """Convert a Sweepable to a list of ParamResolvers."""
-    if sweepable is None:
-        yield ParamResolver({})
-    elif isinstance(sweepable, ParamResolver):
-        yield sweepable
-    elif isinstance(sweepable, Sweep):
-        yield from sweepable
-    elif isinstance(sweepable, dict):
-        yield ParamResolver(cast(Dict, sweepable))
-    elif isinstance(sweepable, Iterable) and not isinstance(sweepable, str):
-        for item in cast(Iterable, sweepable):
-            yield from to_resolvers(item)
-    else:
-        raise TypeError(f'Unrecognized sweepable type: {type(sweepable)}.\n'
-                        f'sweepable: {sweepable}')
+    for sweep in to_sweeps(sweepable):
```

Comment:
I believe that the intuition here is good to merge `to_resolvers` logic with `to_sweeps` as `Sweepable` -> `List[Sweep]` -> `Iterator[ParamResolver]` does make sense to me, and that unifies the behavior. 

However, I think we have to be a bit more careful when merging the logic: 

1. when `sweepable` is a ParamResolver, now it will be converted to a Sweep then back to ParamResolver
2. if it's a Sweep we should just yield paramResolvers from it...